python/cuml/linear_model/logistic_regression.pyx

```diff
@@ -301,10 +301,10 @@ class LogisticRegression(Base, ClassifierMixin):
             )
 
         if self.fit_intercept:
-            self._coef_ = self.qn._coef_[0:-1]
-            self._intercept_ = self.qn._coef_[-1]
+            self._coef_ = self.solver_model._coef_[0:-1]
+            self._intercept_ = self.solver_model._coef_[-1]
```

Comment:
Are these still needed? I think after the fix of PR #2506 they shouldn't be

Reply:
The PR still doesn't handle the fit_intercept case for `coef_` in `qn.pyx` it would return the full `coef_` variable for this. 

https://github.com/rapidsai/cuml/blob/c399db3f7857a23c6b1b2e4cd257f556f2a1e7ee/python/cuml/solvers/qn.pyx#L523

Reply:
should the fit_intercept for coef_ be handled in qn or here?

Reply:
I moved the logic to qn.